Description: Fix FTBFS with GCC 7
 Use NULL 'pointer' instead of end of string for comparison.
Author: Laszlo Boszormenyi (GCS) <gcs@debian.org>
Bug-Debian: https://bugs.debian.org/853559
Last-Update: 2017-09-17

---

--- mozjs24-24.2.0.orig/js/src/shell/jsoptparse.cpp
+++ mozjs24-24.2.0/js/src/shell/jsoptparse.cpp
@@ -253,7 +253,7 @@ OptionParser::extractValue(size_t argc,
     char *eq = strchr(argv[*i], '=');
     if (eq) {
         *value = eq + 1;
-        if (value[0] == '\0')
+        if (value[0] == NULL)
             return error("A value is required for option %.*s", eq - argv[*i], argv[*i]);
         return Okay;
     }
